inginious/frontend/courses.py

```diff
@@ -274,3 +282,11 @@ def get_task_dispenser(self):
     def _build_ac_regex(self, list_ac):
         """ Build a regex for the AC list, allowing for fast matching. The regex is only used internally """
         return re.compile('|'.join(re.escape(x).replace("\\*", ".*") for x in list_ac))
+
+    def is_archive(self):
+        """ Returns true if the course is an archive"""
+        return self._is_archive
+
+    def get_archiving_date(self):
+        """ Returns the date at which the course was archived as a string (None if not archived)"""
+        return datetime.fromisoformat(self._content["archive_date"])
```

Comment:
It seems you do not support a default behaviour in case that key is not existing.